src/stravalib/client.py

```diff
@@ -2043,7 +2042,7 @@ def _fill_buffer(self) -> None:
             except AttributeError:
                 # Entity doesn't have a parse_obj() method, so must be of a
                 # legacy type
-                new_entity = self.entity.deserialize(  # type: ignore[attr-defined]
+                new_entity = self.entity.parse_obj(  # type: ignore[attr-defined]
```

Comment:
parse_obj doesn't take a bind_client kwarg, only a dictionary. i'm actually not sure how this worked previously as is.  @stravalib/stravalib-maintainers can anyone provide me with some direction here? i generally understand the issue but am not sure how to fix it. tests run, mypy fails. so likely we need a new / improved test as well for this case once i understand how the legacy behavior should work? 

thank you!

Reply:
This seems to run if i keep deserialize as is for the legacy behavior. But what i still don't understand is how to recreate legacy behavior to test it.

Reply:
it runs fine with that consolidation! the new_entity object is an activity object with the bound_client assigned. As far as i can see it works as expected and tests continue to run. 

<img width="963" alt="Screenshot 2024-04-17 at 5 50 26 PM" src="https://github.com/stravalib/stravalib/assets/7649194/10b7208d-5130-4000-b31c-c9a18d042f59">

Reply:
this is what i was trying to figure out. i didn't understand what would trigger that except part of the code so i wasn't sure how to test for it.